src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -1096,7 +1096,7 @@ SchemaOwnerName(Oid objectId)
 static bool
 HasMetadataWorkers(void)
 {
-	List *workerNodeList = ActivePrimaryNodeList();
+	List *workerNodeList = ActivePrimaryNodeList(NoLock);
```

Comment:
Reverted this to `NoLock` as the whole metadata story isn't currently well flushed out to call here